### ISSUE TITLE
Evaluating parameter for model trained on Cityscape

### ISSUE THREAD

**Issue by Shankhanil006** [mrharicot/monodepth#145]
Hi,
I want to test the model trained on Cityscapes image on the 200 Kitti disparity maps.In the past issues you have said that for this purpose cityscapes camera parameter should be used which is baseline 0.22 and width to focal 2262 for width of 2048.
So for the evaluation on Kitti scene for which the depth is 1242 approx I changed the width to focal as (2262/2048)*1242 but the results are totally mismatch. 
Could you state the correct formula for converting width to focal for a different width.

Thank You.

**Comment by mrharicot** [comment]
Hi,
Simply replace this line:
https://github.com/mrharicot/monodepth/blob/master/utils/evaluate_kitti.py#L44
with:
`focal_length, baseline = 2262, 0.22`

**Comment by Shankhanil006** [comment]
Hi, 
The line you asked to edit was accessed for eigen split. I am saying about column 5 of table 1 where you tested on kitti split. Did you resize the width of the 200 test images to 2048? When I tested your cityscapes model on the kitti split by changing baseline to .22 and focal to 2262 results are not matching.

Thank You.

**Comment by mrharicot** [comment]
I don't recall what I did exactly for this result, but my guess is that I used the cityscape values only for the pred_depth:
https://github.com/mrharicot/monodepth/blob/master/utils/evaluation_utils.py#L60
should become:
`pred_depth = 2262 * 0.22 / pred_disp`

**Comment by Shankhanil006** [comment]
Tested using the expression you provided but the results are still not coming. Do the focal remains same irrepective of whether the image width is 2048 or 1242?

**Comment by mrharicot** [comment]
Can you share here the results you got for each configuration?

**Comment by Shankhanil006** [comment]
Hi,
1)The result for pred_depth = 2262 * 0.22 / pred_disp and gt_depth = width_to_focal * 0.54 / gt_disp is:
 abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    1.1335,    22.3372,     20.634,      0.762,     94.772,      0.022,      0.068,      0.280
2) when both ground truth and pred depth are calculated using kitti camera parameter:
 abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.7058,    10.4943,     14.665,      0.546,     94.768,      0.054,      0.325,      0.856
This result is nearly equal to the result you provided in your paper.

**Comment by Shankhanil006** [comment]
Hi,
I checked the note you referred to, yes the value will differ a bit . I tested your model trained on Kitti split on the 200 kitti test images and the result are as follows:

 abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.1238,     1.3967,      6.137,      0.217,     30.352,      0.841,      0.937,      0.975
It's very near to your result which may be caused due to the upadtes disparity map..

But for cityscapes the results are nowhere near the value.

**Comment by ehabhelmy82** [comment]
Hello @Shankhanil
did you find out what is the problem?
am facing the same problem as i trained using cityscape only and tested using 

python3 monodepth_main.py --mode test --data_path /data/ehab/data_scene_flow/ --filenames_file ./utils/filenames/kitti_stereo_2015_test_files.txt --log_directory ~/CSd/ --checkpoint_path ~/CSd/CSmodel/model-143600

and evaluated with modified evaluate file
python utils/evaluate_kitti1.py --split kitti --predicted_disp_path ../CSd/CSmodel/disparities.npy --gt_path ../data/data_scene_flow 
where kitti1 has the dollowing modifications:
focal_length, baseline = 2262, 0.22
gt_depth = width_to_focal[width] * 0.22 / (gt_disp + (1.0 - mask))
        pred_depth = 2262 * 0.22 / pred_disp

and got the following results:
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    2.7478,    57.2008,     22.376,      1.303,     62.491,      0.003,      0.008,      0.023


is it closed to results you got?
Thanks

**Comment by ehabhelmy82** [comment]
I changed it as the following:
        gt_depth = width_to_focal[width] * 0.54 / (gt_disp + (1.0 - mask))
and I got the following results
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.5484,     8.2487,     12.573,      0.466,     62.491,      0.158,      0.685,      0.899

is these results like the one you got?

**Comment by Shankhanil006** [comment]
You can see the results I got in the above comments..Your d1_all is way lower than in the paper which is 94 approx.

**Comment by mrharicot** [comment]
@Shankhanil006 from those numbers I am now pretty convinced that I just used the default evaluation code. The difference can be explained by the update of KITTI the ground truth disparity map.

I would also like to point out that these numbers are not very important, we just wanted to show that one cannot expect good qualitative results when only training on a different dataset.

**Comment by Shankhanil006** [comment]
@mrharicot Thank You for your response.  That's must be the case only . And I have not tested on Make3D but there also the result may vary if default evaluation code is used.

**Comment by ehabhelmy82** [comment]
Hi Shankhanil
I used the pretrained model 
tested using cityscapes_test_files.txt 
and evaluated using 
cd /home/ehab/monodepth-master/
python utils/evaluate_kitti1.py --split kitti --predicted_disp_path ../CSPretrained/disparities.npy --gt_path ../data/data_scene_flow
abs_rel,     sq_rel,        rms,    log_rms,     d1_all,         a1,         a2,         a3
    0.9708,    21.9907,     17.116,      0.686,     83.268,      0.089,      0.312,      0.733
is this results make sense?
Here is the modifications I made 
gt_depth = width_to_focal[width] * 0.54 / (gt_disp + (1.0 - mask))
        pred_depth = 2262 * 0.22 / pred_disp

and 

focal_length, baseline = 2262, 0.22
here are attached the two files I used for evaluation attached 
Please help me to fix that if you can
Thanks

**Comment by Shankhanil006** [comment]
@ehabhelmy82 
Hi,
You can see the last comment by @mrharicot  and find that the evaluation code he used for cityscapes is also using Kitti camera parameters not using baseline of 0.22 and focal of 2262 of cityscapes . The result using Kitti camera parameter came to be -

abs_rel, sq_rel, rms, log_rms, d1_all, a1, a2, a3
0.7058, 10.4943, 14.665, 0.546, 94.768, 0.054, 0.325, 0.856

This slight change is result from that in paper is due to the updated KITTI_scene_flow_2015 disparity map.
